sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -118,7 +118,7 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
       val processRowWithoutSerde = if (!ioschema.schemaLess) {
         prevLine: String =>
           new GenericInternalRow(
-            prevLine.split(outputRowFormat).padTo(outputFieldWriters.size, null)
+            prevLine.split(outputRowFormat, -1).padTo(outputFieldWriters.size, null)
```

Comment:
FYI @cloud-fan @maropu For this change, I need to explain is that. 
If we not add parameter -1, 
```
"2@false@Spark SQL@".split('@') will have three elems
"2"
"false"
"Spark SQL"
```
Since we are strict handle data, we need to add -1 the result should be 
```
"2@false@Spark SQL@".split('@', -1) will have three elems
"2"
"false"
"Spark SQL"
""
```